forastero/io.py

```diff
@@ -88,74 +88,137 @@ def __len__(self) -> int:
         return self._width
 
 
+def io_prefix_style(bus: str | None, component: str, role_bus: IORole, role_comp: IORole) -> str:
+    """
+    Style signal names as i/o_(<BUS>_)<COMPONENT> for example i_dma_awaddr and
+    o_dma_awready.
+
+    :param bus:       Name of the bus instance or None if not required
+    :param component: Name of component signal withim the bus
+    :param role_bus:  Interface role of the entire bus instance
+    :param role_comp: Interface role of the component within the bus
+    :returns:         Complete string name
+    """
+    mapping = {
+        (IORole.INITIATOR, IORole.INITIATOR): "o",
+        (IORole.INITIATOR, IORole.RESPONDER): "i",
+        (IORole.RESPONDER, IORole.INITIATOR): "i",
+        (IORole.RESPONDER, IORole.RESPONDER): "o",
+    }
+    full_name = f"{mapping[role_bus, role_comp]}"
+    if bus is not None:
+        full_name += f"_{bus}"
+    return f"{full_name}_{component}"
+
+
+def io_suffix_style(bus: str | None, component: str, bus_role: IORole, comp_role: IORole) -> str:
+    """
+    Style signal names as (<BUS>_)<COMPONENT>_i/o for example dma_awaddr_i and
+    dma_awready_o.
+
+    :param bus:       Name of the bus instance or None if not required
+    :param component: Name of component signal withim the bus
+    :param role_bus:  Interface role of the entire bus instance
+    :param role_comp: Interface role of the component within the bus
+    :returns:         Complete string name
+    """
+    mapping = {
+        (IORole.INITIATOR, IORole.INITIATOR): "o",
+        (IORole.INITIATOR, IORole.RESPONDER): "i",
+        (IORole.RESPONDER, IORole.INITIATOR): "i",
+        (IORole.RESPONDER, IORole.RESPONDER): "o",
+    }
+    full_name = f"{bus}_" if bus is not None else ""
+    return f"{full_name}{component}_{mapping[bus_role, comp_role]}"
+
+
+def io_plain_style(bus: str | None, component: str, bus_role: IORole, comp_role: IORole) -> str:
+    """
+    Style signal names as (<BUS>_)<COMPONENT> for example dma_awaddr and
+    dma_awready.
+
+    :param bus:       Name of the bus instance or None if not required
+    :param component: Name of component signal withim the bus
+    :param role_bus:  Interface role of the entire bus instance
```

Comment:
bus_role and comp_role used in function definition. The description uses role_bus and role_comp